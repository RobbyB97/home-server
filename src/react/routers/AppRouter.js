//Libraries
import React from 'react'
import {BrowserRouter, Route, Switch} from 'react-router-dom'
import { connect } from 'react-redux';
import Loader from 'react-loader-spinner'

//Components
import Header from '../components/Header'
import NavigationMenu from '../components/NavigationMenu'
import DesktopNavigationMenu from '../components/DesktopNavigationMenu'
import LogoBar from '../components/LogoBar'

//Pages
import StyleGuide from '../components/pages/StyleGuide'
import ReactApp from '../components/pages/ReactApp'

// Actions
import {getTitle} from '../actions/title'
import {homeOn, homeOff} from '../actions/home'
import {toggleNavOff} from '../actions/navigation'

// Routes
const AppRouter = ({nav, dispatch, title, home}) => (
  <BrowserRouter>
    <div id={home && "home"}>
      <div id={home && "main-bg"} className="space"></div>
      <div id="header-bg"></div>
      <div id="main-title">
        {}
      </div>

      
      <Header />
      <DesktopNavigationMenu />
      <NavigationMenu />
      <LogoBar></LogoBar>

      <div id="main-content"onClick={() => {
        dispatch(toggleNavOff())
      }}>
        <Switch>
          <Route 
            path="/react/" 
            component={ReactApp} 
            exact={true} 
          />
          <Route 
            path="/react/style_guide/" 
            component={StyleGuide}
          />
        </Switch>
      </div>
    </div>
  </BrowserRouter>
)

const mapStateToProps = (state) => {
  return {
    nav: state.navigation.nav,
    title: state.title.title,
    home: state.home.home
  }
}

export default connect(mapStateToProps)(AppRouter)