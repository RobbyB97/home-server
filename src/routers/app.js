// Libraries
const express = require('express')
const path = require('path')
const hbs = require('hbs')
const chalk = require('chalk')

// Environment variables
const dev = process.env.DEV


// Config
const router = new express.Router()
const http2https = require('../middleware/http2https')


// Homepage
router.get('', http2https, (req, res) => {
  	try {
    	if (dev == 1) {
      		console.log(chalk.blue('Redirecting homepage to react...'))
      		return res.redirect('/react')
    	} 
    	res.redirect('/portfolio')
  	} catch(e) {
    	res.send()
  	}
})

// Portfolio
router.get('/portfolio', http2https, (req, res) => {
  	try {
    	res.render('portfolio/index')
  	} catch(e) {
    	res.send()
  	}
})

module.exports = router
