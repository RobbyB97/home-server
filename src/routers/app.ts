export {}

/* Libraries */
const express = require('express');
const path = require('path');
const chalk = require('chalk');


/* Config */
const router = new express.Router();

const publicPath = path.join(__dirname, '../../../public');
const portfolioRoute = path.join(publicPath, 'portfolio.html');


/* Routes */
router.get('/', (req: any, res: any): any => {
    try {
        return res.sendFile(portfolioRoute);

    } catch(e) {
        res.send();
        console.log(chalk.red('An error occured on the homepage: '));
        console.log(e);
    }
})

router.get('/resume', (req: any, res: any): any => {
    try {
        return res.sendFile(path.join(publicPath, 'resume.pdf'));

    } catch(e) {
        res.send();
        console.log(chalk.red('An error occured on /resume: '));
        console.log(e);
    }
});

router.get('/*', (req, res): any => {
    try {
        return res.sendFile(path.join(publicPath, 'index.html'));

    } catch(e) {
        res.send()
        console.log(chalk.red('An error occured on the wildcard: '));
        console.log(e);
    }
});


module.exports = router;
