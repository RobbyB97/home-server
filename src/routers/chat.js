// Libraries
const express = require('express')
const path = require('path')
const chalk = require('chalk')


// Config
const router = new express.Router()
const http2https = require('../middleware/http2https')


// Routes
router.get('/chat', http2https, (req, res) => {
    try {
        res.render('chat/index')
    } catch(e) {
        res.send()
        console.log(chalk.red('An error occured on /chat: '))
        console.log(e)
    }
})


module.exports = router