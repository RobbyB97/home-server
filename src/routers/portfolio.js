// Libraries
const express = require('express')
const path = require('path')
const chalk = require('chalk')


// Config
const router = new express.Router()
const http2https = require('../middleware/http2https')


// Routes
router.get('/react', http2https, (req, res) => {
    try {
        return res.return('react/index')
    } catch(e) {
        res.send()
        console.log(chalk.red('An error occured on /react: '))
        console.log(e)
    }
})

router.get('/style_guide', http2https, (req, res) => {
    try {
        return res.redirect('/react')
    } catch(e) {
        res.send()
        console.log(chalk.red('An error occured on /react: '))
        console.log(e)
    }
})

router.get('/skills', http2https, (req, res) => {
    try {
        return res.redirect('/react')
    } catch(e) {
        res.send()
        console.log(chalk.red('An error occured on /react: '))
        console.log(e)
    }
})

router.get('/experience', http2https, (req, res) => {
    try {
        return res.redirect('/react')
    } catch(e) {
        res.send()
        console.log(chalk.red('An error occured on /react: '))
        console.log(e)
    }
})


module.exports = router
