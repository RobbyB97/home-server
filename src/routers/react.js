// Libraries
const express = require('express')
const path = require('path')
const chalk = require('chalk')


// Config
const router = new express.Router()
const http2https = require('../middleware/http2https')


// Routes
router.get('/react', http2https, (req, res) => {
    try {
        res.render('react/index')
    } catch(e) {
        res.send()
        console.log(chalk.red('An error occured on /react: '))
        console.log(e)
    }
})

router.get('/react/*', http2https, (req, res) => {
    try {
        res.render('react/index')
    } catch(e) {
        res.send()
        console.log(chalk.red('An error occured on /react: '))
        console.log(e)
    }
})


module.exports = router
