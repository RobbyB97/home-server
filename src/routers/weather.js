// Libraries
const express = require('express')
const path = require('path')
const hbs = require('hbs')

// Config
const router = new express.Router()
const http2https = require('../middleware/http2https')

// Routes
router.get('/weather', http2https, (req, res) => {
  	try {
    	res.render('weather/index', {
      	title: 'Weather App',
      	name: 'Robby'
    })
  	} catch(e) {
    	res.send()
  	}
})

router.get('/weather/about', http2https, (req, res) => {
  	try {
    	res.render('weather/about/index', {
      		title: 'About',
      		name: 'Robby'
    	})
  	} catch(e) {
    	res.send()
  	}
})

router.get('/weather/help', http2https, (req, res) => {
  	try {
    	res.render('weather/help/index', {
      		title: 'Help',
      		name: 'Robby'
    	})
  	} catch(e) {
    	res.send()
  	}
})

router.get('/weather/*', http2https, (req, res) => {
  	try {
    	res.render('weather/404', {
      		title: 'Weather 404',
      		message: 'This page doesn\'t exist :('
    	})
  	} catch(e) {
    	res.send()
  	}
})

module.exports = router
