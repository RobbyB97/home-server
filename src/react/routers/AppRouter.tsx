export {}

/* Libraries */
import * as React from 'react';
import {HashRouter, Route} from 'react-router-dom';

import Portfolio from '../pages/Portfolio';


/* Router */
export class AppRouter extends React.Component {
    render() {
        return (
            <HashRouter>
                <Route path="/" component={Portfolio} exact />
            </HashRouter>
        );
    }
}


export default AppRouter;