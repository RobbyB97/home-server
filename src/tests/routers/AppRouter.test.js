import React from 'react'
import {shallow} from 'enzyme'

import {AppRouter} from '../../react/routers/AppRouter'


test('Render AppRouter', () => {
    const wrapper = shallow(<AppRouter />)

    expect(wrapper).toMatchSnapshot()
})