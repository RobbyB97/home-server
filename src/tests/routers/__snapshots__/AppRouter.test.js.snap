// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render AppRouter 1`] = `
<BrowserRouter>
  <div>
    <Connect(Header) />
    <DesktopNavigation />
    <Connect(MobileNavigation) />
    <LogoBar />
    <div
      id="main-content"
      onClick={[Function]}
    >
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/style_guide"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/skills"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/experience"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/projects"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/resume"
        />
      </Switch>
    </div>
  </div>
</BrowserRouter>
`;
